.github/workflows/lint.yml

```diff
@@ -0,0 +1,26 @@
+name: 'lint'
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+
+jobs:
+  check_types:
+    name: Run eslint
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+
+      - name: Setup Node
+        uses: actions/setup-node@v3
+        with:
+          node-version: 18.x
+          cache: yarn
+
+      - name: Install dependencies
+        run: yarn install
+
+      - name: Run lint
```

Comment:
suggestion: I'm not a GH actions expert, but I believe we can improve this by avoiding the installation of dependencies being done 3 times (types, lint, and test). As this is a small project with simple workflows, I'd have a single workflow file (`build.yml`) with the 3 jobs (types, lint, test) that depend on an initial job called `node_modules`. Then you use `needs: [node_modules]` to declare the chain of jobs. Check [the docs](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-requiring-successful-dependent-jobs).

Reply:
I'm also not an expert haha. Sounds good, I'm going to check it out, thx!

Reply:
it seems it's still failing 🤔 didn't debug it, but you might want to compare it to [another yml of mine](https://github.com/remoteoss/json-schema-form/blob/main/.github/workflows/build.yml) that it's working 👀